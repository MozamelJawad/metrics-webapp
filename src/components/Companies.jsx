import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getCompanies } from '../redux/companies/companiesSlice';

const Companies = () => {
  const [search, setSearch] = useState('');
  const { companies } = useSelector((state) => state.companies);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCompanies());
  }, [dispatch]);

  return (
    <>
      <div className="header">
        <p className="title">Companies</p>
        <input
          className="search"
          onChange={(e) => setSearch(e.target.value)}
          type="text"
          placeholder="Search company ..."
        />
        <div className="header-icon">
          <i className="bi bi-mic-fill" />
          <i className="bi bi-gear-fill" />
        </div>
      </div>
      <div className="cards">
        {
      companies.filter((item) => (search.toLowerCase() === '' ? item : item.symbol.toLowerCase().includes(search))).map((company) => (
        <Link className="link" key={company.symbol} to={`details/${company.symbol}`}>
          <div className="card">
            <i className="bi bi-arrow-right-circle" />
            <span className="company-data">
              <h3 className="symbol">{company.symbol}</h3>
              <p className="founded">{company.founded}</p>
            </span>

          </div>
        </Link>
      ))
}
      </div>
    </>
  );
};

export default Companies;
