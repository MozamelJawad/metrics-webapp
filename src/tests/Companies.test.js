import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Companies from '../components/Companies';
import '@testing-library/jest-dom';

describe('Companies list: ', () => {
  test('Should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Companies />
        </BrowserRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
