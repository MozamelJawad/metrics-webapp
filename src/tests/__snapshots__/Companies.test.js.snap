// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies list:  Should render correctly 1`] = `
<div>
  <div
    class="header"
  >
    <p
      class="title"
    >
      Companies
    </p>
    <input
      class="search"
      placeholder="Search company by name"
      type="text"
    />
    <div
      class="header-icon"
    >
      <i
        class="bi bi-mic-fill"
      />
      <i
        class="bi bi-gear-fill"
      />
    </div>
  </div>
  <div
    class="cards"
  />
</div>
`;
